core-bundle/src/Util/SimpleTokenParser.php

```diff
@@ -137,25 +140,36 @@ private function canUseExpressionLanguage(array $data): bool
 
     private function evaluateExpression(string $expression, array $data, bool $canUseExpressionLanguage): bool
     {
-        if ($canUseExpressionLanguage) {
-            try {
-                return $this->expressionLanguage->evaluate($expression, $data);
-            } catch (\Exception $e) {
-                throw new \InvalidArgumentException($e->getMessage(), 0, $e);
-            }
+        if (!$canUseExpressionLanguage) {
+            return $this->evaluateExpressionLegacy($expression, $data);
+        }
+
+        $unmatchedVariables = array_diff($this->getVariables($expression), array_keys($data));
```

Comment:
We could consider making `getVariables()` a method of our `SimpleTokenExpressionLanguage` and (feature testing the usage with `method_exists`, falling back to `[]`) so that people creating their own EL with custom parsing could handle this individually.

Reply:
> (feature testing the usage with `method_exists`,

Not that big a fan of that.

Reply:
Could also be an interface oc (but maybe overkill for such a niche topic).

Reply:
Is it necessary at all?

Reply:
Only if we'd like to factor the `getVariables` method out of our `SimpleTokenParser` service. It could get necessary if one uses their own expression language that parses things differently. But again, pretty niche. 

I'd vote for leaving it as it is in this PR for now and considering the option if someone ever shows up with a problem.